pkg/api/resource_helpers.go

```diff
@@ -80,12 +81,44 @@ func IsPodReadyConditionTrue(status PodStatus) bool {
 // Extracts the pod ready condition from the given status and returns that.
 // Returns nil if the condition is not present.
 func GetPodReadyCondition(status PodStatus) *PodCondition {
+	_, condition := GetPodCondition(&status, PodReady)
+	return condition
+}
+
+func GetPodCondition(status *PodStatus, conditionType PodConditionType) (index int, condition *PodCondition) {
 	for i, c := range status.Conditions {
-		if c.Type == PodReady {
-			return &status.Conditions[i]
+		if c.Type == conditionType {
+			return i, &status.Conditions[i]
+		}
+	}
+	return -1, nil
+}
+
+// Updates existing pod condition or creates a new one. Sets LastTransitionTime to now if the
+// status has changed.
+// Returns true if pod condition has changed or has been modified.
```

Comment:
nit: "has changed" and "has been modified" mean the same thing, so you could just say "has changed"